Source/Core/VideoCommon/HiresTextures.cpp

```diff
@@ -184,34 +256,52 @@ void HiresTexture::Prefetch()
       }
       if (iter != s_textureCache.end())
       {
-        for (const Level& l : iter->second->m_levels)
+        for (const Level& tex : iter->second->m_levels)
         {
-          size_sum += l.data_size;
+          prefetched_size_sum += tex.data_size;
         }
       }
     }
 
-    if (s_textureCacheAbortLoading.IsSet())
+    if (s_texturePrefetcher_AbortLoading.IsSet())
     {
       return;
     }
 
-    if (size_sum > max_mem)
+    if (prefetched_size_sum >= delegated_max_mem)
     {
       Config::SetCurrent(Config::GFX_HIRES_TEXTURES, false);
 
       OSD::AddMessage(
-          StringFromFormat(
-              "Custom Textures prefetching after %.1f MB aborted, not enough RAM available",
-              size_sum / (1024.0 * 1024.0)),
+          StringFromFormat("Prefetcher: Custom textures prefetching after %.2f MB aborted."
+                           " Maximum delegated memory limit reached at %.2f",
+                           prefetched_size_sum / (1024 * 1024), delegated_max_mem / (1024 * 1024)),
           10000);
       return;
     }
-  }
-  u32 stoptime = Common::Timer::GetTimeMs();
-  OSD::AddMessage(StringFromFormat("Custom Textures loaded, %.1f MB in %.1f s",
-                                   size_sum / (1024.0 * 1024.0), (stoptime - starttime) / 1000.0),
-                  10000);
+
+  }  // End Prefetching
+  prefetch_stoptime_total = Common::Timer::GetTimeMs();
+
+  FileType_Counter_Total_Amount = FileType_Counter_DDS_Amount + FileType_Counter_PNG_Amount +
+                                  FileType_Counter_BMP_Amount + FileType_Counter_TGA_Amount +
+                                  FileType_Counter_JPG_Amount + FileType_Counter_HDR_Amount;
+  std::string FT_C_DDS_s = std::to_string(FileType_Counter_DDS_Amount);
+  std::string FT_C_PNG_s = std::to_string(FileType_Counter_PNG_Amount);
+  std::string FT_C_BMP_s = std::to_string(FileType_Counter_BMP_Amount);
+  std::string FT_C_TGA_s = std::to_string(FileType_Counter_TGA_Amount);
+  std::string FT_C_JPG_s = std::to_string(FileType_Counter_JPG_Amount);
+  std::string FT_C_HDR_s = std::to_string(FileType_Counter_HDR_Amount);
+  std::string FT_C_Total_s = std::to_string(FileType_Counter_Total_Amount);
+  double Pref_sum_total_MB = prefetched_size_sum / (1024.0 * 1024.0);
+  double Pref_total_time_Sec = (prefetch_stoptime_total - prefetch_starttime_total) / 1000.0;
+
+  OSD::AddMessage("Prefetcher: Total of " + FT_C_Total_s + " custom textures loaded, " +
+                      StringFromFormat("%.2f", Pref_sum_total_MB) + " MB in " +
+                      StringFromFormat("%.2f", Pref_total_time_Sec) + " sec, from " + FT_C_DDS_s +
+                      " DDS, " + FT_C_PNG_s + " PNG, " + FT_C_BMP_s + " BMP, " + FT_C_TGA_s +
+                      " TGA, " + FT_C_JPG_s + " JPG, and " + FT_C_HDR_s + " HDR files.",
```

Comment:
I haven't figured that out yet, to do it with as less code replication as possible, will do, but I did had it in mind indeed.